src/index.js

```diff
@@ -20,89 +18,86 @@ const api = (token, options, callback) => {
-				totalToInspect: forks.length
-			});
-			const forkDone = (fork) => {
-				countDoneForks += 1;
-				options.progress({
-					countInspected: countDoneForks,
-					lastInspected: fork.name,
-					totalToInspect: forks.length
-				});
-			};
-
-			// Keep only useless forks
-			async.filter(
-				forks,
-				(fork, filterCallback) => {
-					shouldDeleteFork(github, fork, (error, result) => {
-						if (error) {
-							options.warnings(
-								`Failed to inspect ${fork.name}, skipping`,
-								error
-							);
-							result = false;
-						}
-
-						forkDone(fork);
-						filterCallback(null, result);
-					});
-				},
-				(error, forksToDelete) => {
-					if (error) {
-						return getCallback(error);
-					}
-
-					// Map to our simple objects
-					const response = forksToDelete.map((fork) => ({
-						owner: fork.owner.login,
-						repo: fork.name,
-						url: fork.html_url
-					}));
-					getCallback(null, response);
-				}
-			);
-		})
-		.then(null, (error) => {
-			getCallback(error);
+	// Keep only forks owned by a specified user
+	if (options.user) {
+		forks = forks.filter(({owner}) => owner.login === options.user);
+	}
+
+	let countDoneForks = 0;
+	options.progress({
+		countInspected: 0,
+		totalToInspect: forks.length
+	});
+	const forkDone = (fork) => {
+		countDoneForks += 1;
+		options.progress({
+			countInspected: countDoneForks,
+			lastInspected: fork.name,
+			totalToInspect: forks.length
 		});
+	};
+
+	// Keep only useless forks
+	const processedForks = forks.map(async (fork) => {
+		try {
+			if (await shouldDeleteFork(github, fork)) {
+				// Map to our simple objects
+				return {
+					owner: fork.owner.login,
+					repo: fork.name,
+					url: fork.html_url
+				};
+			}
+		} catch (error) {
+			options.warnings(`Failed to inspect ${fork.name}, skipping`, error);
+		} finally {
+			forkDone(fork);
+		}
+	});
+
+	const forksToDelete = await Promise.all(processedForks);
+	return forksToDelete.filter(Boolean);
 };
 
-api.remove = (token, repos, removeCallback) => {
+const _apiRemove = async (token, repos) => {
 	const github = githubFactory(token);
-	async.each(
-		repos,
-		(repo, callback) => {
-			github.repos.delete({owner: repo.owner, repo: repo.repo}, callback);
-		},
-		removeCallback
+	await Promise.all(
+		repos.map((repo) =>
+			github.repos.delete({owner: repo.owner, repo: repo.repo})
+		)
 	);
 };
 
+/* eslint-disable promise/catch-or-return, promise/prefer-await-to-then, promise/always-return, promise/no-callback-in-promise */
+api.remove = (token, repos, callback) => {
+	_apiRemove(token, repos).then((result) => {
+		callback(null, result);
+	}, callback);
+};
+
+api.get = (token, options, callback) => {
+	if (!callback) {
+		callback = options;
+		options = {};
+	}
+
+	_apiGet(token, options).then((result) => {
+		callback(null, result);
+	}, callback);
+};
+/* eslint-enable promise/catch-or-return, promise/prefer-await-to-then, promise/always-return, promise/no-callback-in-promise */
```

Comment:
This line is required by the linter. If you release a Promise-only version you'll be able to drop this whole block anyway, so I wouldn't worry.